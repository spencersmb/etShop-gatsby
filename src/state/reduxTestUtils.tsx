// this is a handy function that I normally make available for all my tests
import { ICartState } from '@et/types/Cart'
import { IGatsbyConfig } from '@et/types/Gatsby'
import { IProduct } from '@et/types/Products'
import React from 'react'
import initialState from '@redux/reducers/initialState'
import { Provider } from 'react-redux'
import { render } from 'react-testing-library'
// that deal with connected components.
// you can provide initialState or the entire store that the ui is rendered with
import { createStore } from 'redux'

export function renderWithRedux (
	ui: any,
	reducer: any,
	{ state, store = createStore(reducer, state) }: any = initialState
) {
	return {
		...render(<Provider store={store}>{ui}</Provider>),
		// adding `store` to the returned utilities to allow us
		// to reference it in our tests (just try to avoid using
		// this to test implementation details).
		store
	}
}

const testSiteMetaDataQuery = {
	title: `Every-Tuesday Digital Products Shop`,
	description: `Kick off your next, great Gatsby project with this default starter. This barebones starter ships with the main Gatsby configuration files you might need.`,
	author: `@Teelac`,
	authorUrl: 'https://every-tuesday.com/about/#teela',
	siteUrl: `https://shop.every-tuesday.com`,
	siteName: `Every-Tuesday Shop`,
	db: 'http://shopeverytuesday.local',
	route: 'et-shop',
	twitterUrl: 'https://twitter.com/teelacunningham',
	twitterDefaultImage: 'url'
}

export const singleItemQuery: IGatsbyConfig & { wcProduct: IProduct } = {
	site: {
		siteMetadata: {
			...testSiteMetaDataQuery
		}
	},
	wcProduct: {
		type: 'simple',
		date_created_gmt: '2018-09-25T20:35:51',
		date_modified_gmt: '2019-02-18T21:44:25',
		id: '202eca74-fc90-56e7-8269-b59f18a19194',
		name: 'Watercolor texture kit Vol. 1',
		price: '16',
		product_id: 222,
		pwyw: false,
		regular_price: '16',
		sale_price: '',
		slug: 'watercolor-texture-kit-vol-1',
		description: 'description',
		short_description: 'short desc',
		on_sale: false,
		features: {
			description: 'feature desc',
			items: {
				description: 'item desc',
				icon: 'icon',
				title: 'singleItemQuery title'
			}
		},
		categories: [
			{ id: 1, slug: 'fonts', name: 'Fonts' },
			{ id: 2, slug: 'Watercolor', name: 'Watercolor' }
		],
		images: [
			{
				id: 123414,
				alt: 'alt',
				fullSize: {
					url: 'fullsize-url'
				},
				thumbnail: {
					url: 'image-url'
				}
			}
		],
		license: {
			hasExtendedLicense: true,
			type: 'standard',
			extendedItem: {
				slug: 'watercolor-texture-kit-vol-1-ext'
			}
		},
		seo: {
			desc: 'Add serious vibrant color and detail with this kit packed with 32 unique watercolor textures, 12 seamless, repeatable watercolor texture patterns and 2 bonus watercolor paper patterns.',
			title: 'Watercolor texture kit Vol. 1'
		},
		tags: [
			{ id: 1, name: 'Texture Kit', slug: 'texture-kit' },
			{ id: 2, name: 'Waterbrush', slug: 'waterbrush' }
		],
		localFile: {
			name: 'file name',
			id: '1234',
			childImageSharp: {
				fluid: {
					src: 'src'
				},
				fixed: {
					width: '400',
					height: '400',
					src: 'src'
				}
			}
		}
	}
}

export const testCartEmpty: ICartState = {
	couponCode: {
		code: 'string',
		discount: 'string',
		loading: false,
		product_ids: [],
		submitted: false,
		type: 'string',
		valid: false,
	},
	isOpen: false,
	items: {
	},
	loaded: true,
	paymentType: 'Stripe',
	totalItems: 0,
	totalPrice: 0
}
