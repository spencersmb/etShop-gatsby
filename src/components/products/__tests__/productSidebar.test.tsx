import SideBar from '@components/products/modules/productDetailsSidebar'
import { ProductKey, testProducts } from '@redux/reduxTestUtils'
import React from 'react'
import renderer from 'react-test-renderer'
import {
	render,
	cleanup
} from 'react-testing-library'
import 'jest-dom/extend-expect'

afterEach(cleanup)

const propsDefault = {
	details:
		{
			file_types: ['jpg'],
			file_size: '1.2 Gigs',
			dpi: '600',
			programs: [
				'photoshop', 'procreate', 'illustrator'
			],
			reqs: false
		},
	fontPreview: false,
	onChange: jest.fn(),
	isStandardLicense: true,
	licenses: testProducts[ProductKey.WatercolorStd].product_licenses
}

const webfontDefault = {
	details:
		{
			file_types: ['otf', 'ttf', 'woff', 'woff2'],
			file_size: '1.2 Gigs',
			dpi: '600',
			programs: [
				'photoshop', 'procreate', 'illustrator'
			],
			reqs: false
		},
	fontPreview: true,
	onChange: jest.fn(),
	isStandardLicense: true,
	licenses: testProducts[ProductKey.WatercolorStd].product_licenses
}
const webfontExt = {
	details:
		{
			file_types: ['otf', 'ttf', 'woff', 'woff2'],
			file_size: '1.2 Gigs',
			dpi: '600',
			programs: [
				'photoshop', 'procreate', 'illustrator'
			],
			reqs: false
		},
	fontPreview: true,
	onChange: jest.fn(),
	isStandardLicense: false,
	licenses: testProducts[ProductKey.WatercolorStd].product_licenses
}
describe('Product Description', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(
				<SideBar {...propsDefault}/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('Should render correct file types for std item', () => {
		const modalRender = render(<SideBar {...propsDefault}/>)
		expect(modalRender.getByTestId('fileTypes').children.length).toEqual(1)
	})

	// count # of children to determine if it rendered
	it('Should not render web font upgrade message', () => {
		const modalRender = render(<SideBar {...propsDefault}/>)
		expect(modalRender.getByTestId('fileTypeSection').children.length).toEqual(2)
	})

	it('Should render resolution dpi message', () => {
		const modalRender = render(<SideBar {...propsDefault}/>)
		expect(modalRender.getByTestId('dpi').innerHTML).toEqual(`600<span>dpi</span>`)
	})

	it('Should render file size message', () => {
		const modalRender = render(<SideBar {...propsDefault}/>)
		expect(modalRender.getByTestId('fileSize').innerHTML).toEqual(`1.2<span>Gigs</span>`)
	})

	it('Should render correct number of programs', () => {
		const modalRender = render(<SideBar {...propsDefault}/>)
		expect(modalRender.getByTestId('programs').children.length).toEqual(3)
	})

	// For WEB FONTS
	it('Should render web font upgrade message', () => {
		const modalRender = render(<SideBar {...webfontDefault}/>)
		expect(modalRender.getByTestId('upgrade').innerHTML).toEqual('<p>Web Font? Upgrade to</p><span>Extended License</span>')
	})

	it('Should not render web font types with standard license', () => {
		const modalRender = render(<SideBar {...webfontDefault}/>)
		expect(modalRender.getByTestId('fileTypes').children.length).toEqual(2)
	})

	it('Should call clickHandler on click', () => {
		const modalRender = render(<SideBar {...webfontDefault}/>)
		const button = modalRender.getByTestId('upgrade')
		button.click()
		expect(webfontDefault.onChange).toHaveBeenCalledTimes(1)
	})

	it('Should render web font types with extended license selected', () => {
		const modalRender = render(<SideBar {...webfontExt}/>)
		expect(modalRender.getByTestId('fileTypes').children.length).toEqual(4)
	})

	it('Should not render web font message', () => {
		const modalRender = render(<SideBar {...webfontExt}/>)
		expect(modalRender.queryByTestId('upgrade')).toBeNull()
	})

})
