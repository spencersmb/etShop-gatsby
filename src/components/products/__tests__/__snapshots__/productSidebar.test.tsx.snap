// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Description renders correctly 1`] = `
<aside
  className="sc-gZMcBi iARPnT"
>
  <div
    className="sc-gqjmRU eTvGQB"
  >
    <div
      className="sc-iwsKbI jlWZrW"
    >
      <test-file-stub />
    </div>
    <div
      className="sc-EHOje gWPvZT"
    >
      <h3
        className="sc-dnqmqq fpYqNh"
      >
        DETAILS
      </h3>
    </div>
    <div
      className="sc-EHOje gWPvZT"
      data-testid="fileTypeSection"
    >
      <h6
        className="sc-bZQynM hlAzNT"
      >
        FILE TYPES
      </h6>
      <div
        className="sc-ifAKCX iBPxhH"
        data-testid="fileTypes"
      >
        <div
          className="sc-bxivhb ksBBKm"
        >
          jpg
        </div>
      </div>
    </div>
    <div
      className="sc-EHOje gWPvZT"
    >
      <div
        className="sc-gzVnrw ioMsCW"
      >
        <test-file-stub />
      </div>
      <h6
        className="sc-bZQynM hlAzNT"
      >
        RESOLUTION
      </h6>
      <div
        className="sc-htoDjs cYYHFH"
        data-testid="dpi"
      >
        600
        <span>
          dpi
        </span>
      </div>
    </div>
    <div
      className="sc-EHOje gWPvZT"
    >
      <div
        className="sc-gzVnrw ioMsCW"
      >
        <test-file-stub />
      </div>
      <h6
        className="sc-bZQynM hlAzNT"
      >
        File Size
      </h6>
      <div
        className="sc-htoDjs cYYHFH"
        dangerouslySetInnerHTML={
          Object {
            "__html": "1.2<span>Gigs</span>",
          }
        }
        data-testid="fileSize"
      />
    </div>
    <div
      className="sc-EHOje gWPvZT"
    >
      <div
        className="sc-gzVnrw ioMsCW"
      >
        <test-file-stub />
      </div>
      <h6
        className="sc-bZQynM hlAzNT"
      >
        WORKS WITH
      </h6>
      <ul
        className="sc-bwzfXH gwZJzL"
        data-testid="programs"
      >
        <li>
          <test-file-stub />
        </li>
        <li>
          <img
            alt="Procreate"
            src="/logos/procreateLogo.png"
          />
        </li>
        <li>
          <test-file-stub />
        </li>
      </ul>
    </div>
  </div>
</aside>
`;
